File: Programmers/쇠막대기.cpp
#include <string>
#include <vector>
#include <stack>
#include <functional>
using namespace std;

int solution(string str) {
	int ans = 0;
	stack<char> s;
	for (int i = 0; i < str.length(); ++i) {
		if (str[i] == '(') s.push('(');
		else {
			s.pop();
			if (str[i - 1] == ')') ans += 1;
			else ans += s.size();
		}
	}
	return ans;
}

File: Programmers/쇠막대기.py
def solution(arrangement):
    s = []
    answer = 0
    prev = None
    for x in arrangement:
        if x is '(':
            s.append(0)
        else:
            s.pop()
            answer += 1 if prev is ')' else len(s)
        prev = x

    return answer